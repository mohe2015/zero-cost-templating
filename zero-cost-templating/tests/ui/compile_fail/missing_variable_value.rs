#![deny(warnings)]
#![feature(async_closure, async_iterator, coroutines, gen_blocks, noop_waker)]

extern crate alloc;

use alloc::borrow::Cow;

use zero_cost_templating::yields;
use zero_cost_templating_macros::template_stream;

#[template_stream("test.html.hbs")]
pub gen fn test() -> Cow<'static, str> {
    let template = yields!(test0());
    let page_title = Cow::from("thetitle");
    let template = yields!(template.page_title0());
    let csrf_token = Cow::from("thetoken");
    let template = yields!(template.csrf_token1(csrf_token));
    let template = yields!(template.copyright_year4(Cow::from("2023".to_string())));
}

pub fn main() {}
