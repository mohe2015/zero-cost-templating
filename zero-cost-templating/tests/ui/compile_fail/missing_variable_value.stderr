error[E0061]: this method takes 1 argument but 0 arguments were supplied
  --> tests/ui/compile_fail/missing_variable_value.rs:15:29
   |
15 |     let template = template.test_page_title1();
   |                             ^^^^^^^^^^^^^^^^-- an argument is missing
   |
note: method defined here
  --> tests/ui/compile_fail/missing_variable_value.rs:10:1
   |
10 | #[template_stream("test.html.hbs")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the attribute macro `template_stream` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
   |
15 |     let template = template.test_page_title1(/* page_title */);
   |                                             ~~~~~~~~~~~~~~~~~~
